NumericOutcomeCompression.md

```diff
@@ -101,261 +101,265 @@ these integers (interpreted in base `B`) where all digits not used may be any va
-        if (prefixDigits.nonEmpty) {
-            Vector(prefixDigits) // Total Optimization
-        } else {
-            throw new IllegalArgumentException("DLCs with only one outcome are not supported.")
+        // If start and end are empty or start is all zeros and end is all (base - 1), then the prefix can cover the original interval
+        if start.iter().all(|x| *x == 0    if (ds.is_empty() && de.is_empty())
+           || (ds.iter().all(|x| *x == 0) && de.iter().all(|x| *x == base - 1))
+        {
+            return vec![prefix];
+        }
+
+        // We can remove the trailing 0s from the start and trailing base - 1 from the end
+        // as they will be covered the interval represented by the digits in front of them.
+        remove_tail(&mut ds, 0);
+        remove_tail(&mut de, base - 1);
+
+        // We initialize the stack with the start digits.
+        let mut stack = ds.clone();
+        let mut list = Vec::new();
+
+        // This will generate all the prefixes for the interval [start, start[0]..base - 1]. E.g.
+        // if start is 1234 in base 10, this will generate for [1234, 1999].
+        while stack.len() != 1 {
+            let i = stack.len() - 1;
+            // Once the last digit of the stack is base - 1, we can save the prefix and pop a digit.
+            // E.g. if we have our stack as [1, 2, 3, 9], next is [1, 2, 4, 0], but we don't need the last 0
+            // as we can cover with [1, 2, 4].
+            if stack[i] == base - 1 {
+                list.push(stack.clone());
+                stack.pop();
+                // We can remove any base - 1 digits at this point. E.g. if we had [1, 2, 9, 9] above,
+                // now we have [1, 2, 9], next is [1, 3, 0], but similarly as above we can get rid of
+                // the trailing zero.
+                remove_tail(&mut stack, base - 1);
+                // We increment the last digit (e.g. move from [1, 2] to [1, 3] in the example above).
+                let j = stack.len() - 1;
+                stack[j] += 1;
+            } else if stack[i] == 0 {
+                // We can always get rid of trailing zeros (up to the first digit). E.g. if we have
+                // out stack as [1, 3, 0, 0], [1, 3] is enough to cover the interval
+                remove_tail(&mut stack, 0);
+            } else {
+                // We save the stack an increment the last digit. E.g. if we had [1, 2, 3, 4], we save it
+                // and move to [1, 2, 3, 5].
+                list.push(stack.clone());
+                stack[i] += 1;
+            }
+            assert!(stack.iter().all(|x| x < &base));
         }
-    } else if (prefixDigits.length == numDigits - 1) { // Special Case: Front Grouping = Back Grouping
-        startDigits.last.to(endDigits.last).toVector.map { lastDigit =>
-            prefixDigits :+ lastDigit
+
+        // All the single digits in ]start[0]; end[0][ are sufficient to cover their respective intervals.
```

Comment:
> `]start[0]; end[0][`

I think this is meant to be `[start[0]; end[0]]`

Reply:
I think `] start[0]; end[0] [` is correct. If it were  `[start[0]; end[0]]` then in the example it would include `1` and `4` which are covered in the loops above and under. Let me know if maybe something is unclear (or if maybe I'm just missing something).